packages/@react-aria/dnd/src/useDrop.ts

```diff
@@ -311,23 +313,34 @@ export function useDrop(options: DropOptions): DropResult {
   });
 
   let {ref} = options;
-  useLayoutEffect(() => DragManager.registerDropTarget({
-    element: ref.current,
-    getDropOperation: getDropOperationKeyboard,
-    onDropEnter(e) {
-      setDropTarget(true);
-      onDropEnter(e);
-    },
-    onDropExit(e) {
-      setDropTarget(false);
-      onDropExit(e);
-    },
-    onDrop: onKeyboardDrop,
-    onDropActivate
-  }), [ref, getDropOperationKeyboard, onDropEnter, onDropExit, onKeyboardDrop, onDropActivate]);
+  useLayoutEffect(() => {
+    if (isDisabled) {
+      return;
+    }
+    return DragManager.registerDropTarget({
+      element: ref.current,
+      getDropOperation: getDropOperationKeyboard,
+      onDropEnter(e) {
+        setDropTarget(true);
+        onDropEnter(e);
+      },
+      onDropExit(e) {
+        setDropTarget(false);
+        onDropExit(e);
+      },
+      onDrop: onKeyboardDrop,
+      onDropActivate
+    });
+  }, [isDisabled, ref, getDropOperationKeyboard, onDropEnter, onDropExit, onKeyboardDrop, onDropActivate]);
 
   let {dropProps} = useVirtualDrop();
-  
+  if (isDisabled) {
+    return {
+      dropProps: {},
+      dropButtonProps: {disabled: true},
```

Comment:
I think we want this instead:

```suggestion
      dropButtonProps: {isDisabled: true},
```

Reply:
By setting `disabled` here, I made sure the DropZone can't be focused with the tab key when it's disabled. Other components like NumberField, Switch, and Button don't get tab focus when they're disabled. What do you think?

Reply:
Using the `useButton` in DropZone based on `dropButtonProps` might be a viable approach. I'll give it a try.

Reply:
I've fixed it in the following commit:
[Fix dropButtonProps using useButton hooks](https://github.com/adobe/react-spectrum/pull/5958/commits/3518f02f767718479aeae54934700e6dc6575d07)